### ISSUE TITLE
test_msa failing without en_US locale

### ISSUE THREAD

**Issue by magnunor** [hyperspy/hyperspy#1717]
As mentioned in https://github.com/hyperspy/hyperspy/issues/455#issuecomment-322049259, I've been having some trouble with getting the msa tests to pass on Debian unstable.

Seems like this is due to the en_US locale missing, and generating it on the system fixes the issue.

```python
python3 -m pytest hyperspy/tests/io/test_msa.py
```

Error message: `E           UserWarning: I couldn't read the date information due toan unexpected error. Please report this error to the developers`

I fix this by adding `en_US.UTF-8 UTF-8` to `/etc/locale.gen`, then running `locale-gen`

**Comment by magnunor** [comment]
The problem is line https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/hyperspy/io_plugins/msa.py#L227, which tries to set the locale to en_US:

```Python
locale.setlocale(locale.LC_TIME, ('en_US', 'utf8'))
```

However, that locale is not necessarily there, giving the error `Error: unsupported locale setting`.

This also happens on my Debian desktop install.

**Comment by ericpre** [comment]
The US locale to needed to read correctly the msa files, see https://github.com/hyperspy/hyperspy/pull/1240#r7402689.

However, it is an [`try: except:` block](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/hyperspy/io_plugins/msa.py#L225), so it should catch the error. By the way, the [warning message](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/hyperspy/io_plugins/msa.py#L243) could be more explicit!

**Comment by ericpre** [comment]
It seems that the actual issue here is that the warning is considered as an error! The msa code works as expected.
It may worth trying to put the warning to the logger instead of using `warnings.warn` or an alternative is to change the parameter during the build which turns the warning as an error (I don't know if this is possible)

**Comment by ericpre** [comment]
Fixed in #2171.
